### ISSUE TITLE
Model validation uses the incorrect class labels

### ISSUE THREAD

**Issue by Shahab-Ahmed** [ultralytics/ultralytics#18872]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I trained an object detection model on apples and oranges; however, when I run the validation to get some graphs it seems to override the class names from apple and orange to person and bicycle.

It seems to be changing from my custom labels to the first two labels from the COCO dataset dataset and I cannot manually alter it to use my class names

Cell: print(model.names) # direcly after model training
Output: {0: 'apple', 1: 'citrus'}

Cell: from ultralytics import YOLO

model = YOLO("/content/yolo11m.pt")

print(model.names)

Output: {0: 'person', 1: 'bicycle', 2: 'car', 3: 'motorcycle', 4: 'airplane', 5: 'bus', 6: 'train', 7: 'truck', 8: 'boat', 9: 'traffic light', 10: 'fire hydrant', 11: 'stop sign', 12: 'parking meter', 13: 'bench', 14: 'bird', 15: 'cat', 16: 'dog', 17: 'horse', 18: 'sheep', 19: 'cow', 20: 'elephant', 21: 'bear', 22: 'zebra', 23: 'giraffe', 24: 'backpack', 25: 'umbrella', 26: 'handbag', 27: 'tie', 28: 'suitcase', 29: 'frisbee', 30: 'skis', 31: 'snowboard', 32: 'sports ball', 33: 'kite', 34: 'baseball bat', 35: 'baseball glove', 36: 'skateboard', 37: 'surfboard', 38: 'tennis racket', 39: 'bottle', 40: 'wine glass', 41: 'cup', 42: 'fork', 43: 'knife', 44: 'spoon', 45: 'bowl', 46: 'banana', 47: 'apple', 48: 'sandwich', 49: 'orange', 50: 'broccoli', 51: 'carrot', 52: 'hot dog', 53: 'pizza', 54: 'donut', 55: 'cake', 56: 'chair', 57: 'couch', 58: 'potted plant', 59: 'bed', 60: 'dining table', 61: 'toilet', 62: 'tv', 63: 'laptop', 64: 'mouse', 65: 'remote', 66: 'keyboard', 67: 'cell phone', 68: 'microwave', 69: 'oven', 70: 'toaster', 71: 'sink', 72: 'refrigerator', 73: 'book', 74: 'clock', 75: 'vase', 76: 'scissors', 77: 'teddy bear', 78: 'hair drier', 79: 'toothbrush'}

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Shahab-Ahmed, thank you for bringing this to our attention 🚀! We recommend exploring the [Ultralytics Docs](https://docs.ultralytics.com/) for guidance, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples. These resources may help address common challenges quickly.

If this is a 🐛 Bug Report, it would be immensely helpful if you could provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us understand and debug the issue effectively. For example, please share the steps you took between training and validation, as well as any relevant configurations, dataset details, or code snippets.

In the meantime, ensure you're using the latest version of the `ultralytics` package to avoid issues that might have already been resolved. You can upgrade using the following command:

```bash
pip install -U ultralytics
```

Additionally, make sure your environment meets all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), including running Python>=3.8 and PyTorch>=1.8.

## Debugging Steps

Validate your YOLO environment to ensure the class labels are being loaded correctly:
1. After training your custom model, confirm your `model.names` directly matches your dataset classes (`print(model.names)`).
2. Check for any inadvertent overwrites or mismatches when loading your trained model for validation.
3. Run your validation scripts and verify all settings are correctly aligned with your trained model configuration files.

For collaborative troubleshooting and real-time discussions, feel free to join:
- 🎧 [Discord](https://discord.com/invite/ultralytics) 
- 💬 [Discourse](https://community.ultralytics.com/) 
- 📚 [Reddit](https://reddit.com/r/Ultralytics)

## Run Environments

To ensure consistency, you can validate your models in one of our verified YOLO environments:
- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>.
- **Cloud**: Utilize **Google Cloud** or **Amazon's** prebuilt Deep Learning AMIs.
- **Docker**: Use the official YOLO Docker image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>.

## CI Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

This badge indicates the status of our [continuous integration (CI) tests](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule). If green, all YOLO modes and tasks are running correctly across all supported platforms.

Please note that this is an automated response 🤖 to provide helpful details immediately, but an Ultralytics engineer will review your issue and assist further as soon as possible. Thank you for your patience and for contributing to the Ultralytics community! ✨

**Comment by glenn-jocher** [comment]
@Shahab-Ahmed yolo11m.pt is the official COCO-pretrained model. Your trained model is in your runs/train directory.

**Comment by Y-T-G** [comment]
You should load the best.pt or last.pt

**Comment by Shahab-Ahmed** [comment]
> [@Shahab-Ahmed](https://github.com/Shahab-Ahmed) yolo11m.pt is the official COCO-pretrained model. Your trained model is in your runs/train directory.

Ok thanks that makes sense.

**Comment by glenn-jocher** [comment]
You're welcome! Let us know if you have any further questions or need additional assistance with your model.
